src/utils/cache.py

```diff
@@ -0,0 +1,70 @@
+"""Utility class containing instrument-specific information. This information
+is used to generate domain-valid currencies, ticker symbols, exchange codes,
+countries of origin, and any combination thereof such as RICs
+"""
+import pandas as pd
+
+
+class Cache:
+    """
+    A Class acting as a remote store for instrument-specific information.
+    Created to keep track of which combinations of the hard-coded values have
+    already been generated with the sole purpose of avoiding duplication.
+
+    Attributes
+    ----------
+    cache : dict
+        Stores specified key, value(s) pairs. Primarily used to keep track of
+        what prior generations have yielded to avoid duplications.
+        Further use to store domain-specific data such as that listed.
+
+    Methods
+    -------
+    persist_to_cache(field_name, field_value)
+        Stores the given value in the pseudo-cache(dict) under the given key
+
+    retrieve_from_cache(field_name)
+        Returns the value stored under given key
+
+    """
+
+    def __init__(self):
+        self.__cache = {}
+        self.persist_to_cache('currencies', ['USD', 'CAD', 'EUR', 'GBP'])
+        self.persist_to_cache('tickers', pd.read_csv('tickers.csv')['Symbol']
+                              .drop_duplicates().values.tolist())
+        self.persist_to_cache('exchange_codes', ['L', 'N', 'OQ', 'SI', 'AL',
```

Comment:
Refer to earlier comment about Exchange domain object with exchange code, country of issue and currency